packages/builder-vite/plugins/react-docgen.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  parse,
+  handlers as docgenHandlers,
+  resolver as docgenResolver,
+  importers as docgenImporters,
+} from 'react-docgen';
+import type { DocumentationObject } from 'react-docgen/lib/Documentation';
+import type { Plugin } from 'vite';
+import actualNameHandler from './docgen-handlers/actualNameHandler';
+
+type DocObj = DocumentationObject & { actualName: string };
+
+// TODO: None of these are able to be overridden, so `default` is aspirational here.
+const defaultHandlers = Object.values(docgenHandlers).map((handler) => handler);
+const defaultResolver = docgenResolver.findAllExportedComponentDefinitions;
+const defaultImporter = docgenImporters.makeFsImporter();
+const handlers = [...defaultHandlers, actualNameHandler];
+
+export function reactDocgen(): Plugin {
+  return {
+    name: 'react-docgen',
+    enforce: 'pre',
+    async transform(src: string, id: string) {
+      if (/\.(mjs|tsx|jsx)$/.test(id)) {
```

Comment:
what about `.js?`

Reply:
When using vite, it's required to use .jsx or .tsx for JSX syntax.  But, I guess technically it's possible to create a react component using `react.createElement()`, so it doesn't hurt to check non-jsx files too.  I'll change it.